flatcar-linux/bootengine
# disk-uuid: improve logic for UUID randomization

# Move UUID randomization code to its own script

We're removing the code in GRUB that detects whether the disk needs to be randomized or not (flatcar-linux/scripts#82). So, we now need to detect whether the change needs to happen during initramfs.

Before, `ignition-generator` was trying to detect whether the `disk-uuid` unit should run or not. Due to ordering and timing issues, detecting whether there's a disk with UUID 00000000-0000-0000-0000-000000000001 in `ignition-generator` doesn't work.  So, instead, move the logic to a separate script, that checks whether something needs to be done or not, and execute the unit unconditionally. The unit verifies if it needs to randomize the UUID and only does so when necessary

# How to use / Testing done

Building an image with this change plus the change in flatcar-linux/scripts#82 leads to GRUB booting successfully on a c3.medium.x86 machine followed by the disk UUID getting randomized.

WIP notice: Due to issues with current flatcar-master-alpha, I wasn't yet able to fully test this change on all platforms, I only manually tested it on Packet. Once alpha is fixed, I'll test again on all platforms.
